docs/source/advanced_config.rst

```diff
@@ -166,21 +164,19 @@ The C preprocessor looks for the output of this step by default.
 If not found, it will fall back to looking for .c files in the source listing.
 
 .. code-block::
+    :linenos:
 
-        steps = [
-            ...
-            CPragmaInjector(),
-            c_preprocessor(),
-            ...
-        ]
-
+    ...
+    c_pragma_injector(config)
+    preprocess_c(config)
+    ...
 
 Custom Steps
 ============
-If you need a custom build step, you can create a subclass of the :class:`~fab.steps.Step` class.
+If you need a custom build step, you can create a function with the @step_timer decorator.
```

Comment:
For later, a better name which doesn't mention timers? It doesn't seem like a useful concept to include and is potentially confusin.